paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/ResourceFolderRepository.kt

```diff
@@ -75,4 +87,127 @@ class ResourceFolderRepository(
     if (namespace != this.namespace) null else resourceTable[resourceType]
 
   override fun getNamespace(): ResourceNamespace = namespace
+
+  /**
+   * Inserts the given resources into this repository.
+   */
+  private fun commitToRepository(itemsByType: Map<ResourceType, ListMultimap<String, ResourceItem>>) {
+    for ((key, value) in itemsByType) {
+      getOrCreateMap(key).putAll(value)
+    }
+  }
+
+  private fun getOrCreateMap(type: ResourceType): ListMultimap<String, ResourceItem> =
+    // Use LinkedListMultimap to preserve ordering for editors that show original order.
+    resourceTable.computeIfAbsent(type) { LinkedListMultimap.create() }
+
+  private class Loader(
+    private val repository: ResourceFolderRepository
+  ) : RepositoryLoader<ResourceFolderRepository>(
+    resourceDirectoryOrFile = repository.resourceDir.toPath(),
+    resourceFilesAndFolders = null,
+    namespace = repository.namespace
+  ) {
+    private val resources = EnumMap<ResourceType, ListMultimap<String, ResourceItem>>(ResourceType::class.java)
+    private val sources = linkedMapOf<PathString, ResourceItemSource<BasicResourceItem>>()
```

Comment:
Use LinkedHashMap rather than HashMap because HashMap is unordered, so unit tests results are different on different OS. Use LinkedHashMap to keep original inserting order to achieve consistent test results

Reply:
You might want to restrict the property type to MutableMap or use `mutableMapOf` which also is a linked hashmap.